File: server/src/main/java/message/Message.java
package message;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public abstract class Message {
    public long generatedTime;

    protected Message() {
        generatedTime = System.currentTimeMillis();
    }

    public String messageType() {
        return this.getClass().getSimpleName();
    }
}

File: Client/Fruitster/Assets/Script/NetConnection/Messages/Message.cs
using System;
using System.Net;
using Script.Utils;
using UnityEngine.Serialization;

namespace Script.Messages
{
    [Serializable]
    public abstract class Message
    {
        public long generatedTime;
        
        protected Message()
        {
            generatedTime = TimeUtils.CurrentTimeMillis();
        }

        public String GetMessageType()
        {
            return GetType().Name;
        }
    }
}